src/useObservable.ts

```diff
@@ -3,6 +3,14 @@ import { useRef } from "react"
 
 type SupportedValues = object | Map<any, any> | any[]
 
-export function useObservable<T extends SupportedValues>(initialValue: T): T {
-    return useRef(observable(initialValue)).current
+export function useObservable<T extends SupportedValues>(initialValue: T | (() => T)): T {
+    const observableRef = useRef((null as unknown) as T)

```

Comment:
What about this way? Feels awkward double typing like that.
```suggestion
    const observableRef = useRef<T | null>(null)
```